tests/suites/test_suite_x509write.function

```diff
@@ -377,12 +377,19 @@ void x509_crt_check( char *subject_key_file, char *subject_pwd,
     if( pk_wrap == 2 )
         TEST_ASSERT( mbedtls_pk_get_type( &issuer_key ) == MBEDTLS_PK_OPAQUE );
 
+#if !defined(MBEDTLS_BIGNUM_C)
     TEST_ASSERT( mbedtls_test_read_mpi( &serial, serial_str ) == 0 );
+#endif
 
     if( ver != -1 )
         mbedtls_x509write_crt_set_version( &crt, ver );
 
+#if !defined(MBEDTLS_BIGNUM_C)
     TEST_ASSERT( mbedtls_x509write_crt_set_serial( &crt, &serial ) == 0 );
+#else
+    TEST_ASSERT( mbedtls_x509write_crt_set_serial_new( &crt, serial_str,
```

Comment:
This works with the current API of the function, but no longer will with the new API. I suggest changing the type of this parameter to `data_t seri_argal` and then we can use `serial_arg->x, serial_arg->len` here. (And for the "old" path we'll just use `mbedtls_mpi_read_binary()` instead.)

(Here I'm using `serial_arg` to avoid collision with `serial` that's currently an MPI, but I think this one should be called `serial` and the other `serial_mpi`.)